### ISSUE TITLE
merge/compare with bib.jl

### ISSUE THREAD

**Issue by stevengj** [bramtayl/BibTeX.jl#2]
Would be good to compare with https://github.com/jacobcvt12/bib.jl for correctness and performance; cc @jacobcvt12.

Would also be good to compare performance (and features) with Python bibTeX parsers like https://github.com/ptigas/bibpy or https://github.com/sciunto-org/python-bibtexparser, or the C parser http://search.cpan.org/~gward/btparse-0.34/doc/btparse.pod or http://www.gerd-neugebauer.de/software/TeX/BibTool/en/

**Comment by bramtayl** [comment]
I think I might be a bit farther along than bib.jl. Is performance a priority? I think it might be faster to figure out how to use a lazy iterator, especially if it could move directly through a file (it looks like `Base.RegexMatchIterator` requires an in memory string).

**Comment by stevengj** [comment]
No, performance doesn't seem like a priority for this sort of thing, but it would be interesting to know how it compares to e.g. `bibpy`, which also uses regexes.

**Comment by jacobcvt12** [comment]
agreed that this is much further along than `bib.jl`. I only briefly outlined the package before being away from the computer all weekend, so there are no comparisons to make at this point. With regard to my question on discourse  (https://discourse.julialang.org/t/citations-in-docs/5066/17), I'll continue the discussion there.
